STM32U3: add UART DMA test

kbidani opened the pull request:
Add an overlay for tests/drivers/uart/uart_async_api/ for these two boards to test UART with DMA.

kbidani committed:
tests: drivers: uart: async_api: add stm32u3 boards overlays

Add UART test overlays for Nucleo U385RG-Q board.
Remove non serial boot conf file since they are now unnecessary.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kbidani commented:
rebased on gpdma branch.

djiatsaf-st commented on tests/drivers/uart/uart_async_api/boards/nucleo_u385rg_q.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_NOCACHE_MEMORY=y
+CONFIG_TEST_USERSPACE=n
```
I think these configs not required for this board.

djiatsaf-st reviewed (commented):
Nit: reoder commits 

dts: arm: st: u3: add gpdma on top  of boards: st: stm32u3: enable dma for stm32u3 boards

Otherwise LGTM

kbidani committed:
boards: st: stm32u3: update documentation for stm32u3 boards

update documentation to indicate that UART is
available on Nucleo U385RG-Q board.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

etienne-lms reviewed (commented):
LGTM aside rebase requirement.

dkalowsk commented:
@kbidani needs a local rebase, please address so we can get this in for 4.2

kartben commented:
@kbidani please rebase
